File: appmonet-sdk/appmonet-sdk-core/src/main/java/com/monet/bidder/AdServerBannerListener.java
package com.monet.bidder;

import android.view.View;

public interface AdServerBannerListener {
//    protected static final Logger sLogger = new Logger("BannerListener");

  void onAdClosed();

  void onAdOpened();

  boolean onAdLoaded(final View view);

  void onAdClicked();

  void onAdError(ErrorCode errorCode);

  void onAdRefreshed(View view);

  enum ErrorCode {
    NO_FILL,
    INTERNAL_ERROR,
    TIMEOUT,
    UNKNOWN,
    BAD_REQUEST
  }
}

File: appmonet-sdk/appmonet-sdk-core/src/main/java/com/monet/bidder/AdServerBannerListener.kt
package com.monet.bidder

import android.view.View

interface AdServerBannerListener {
  fun onAdClosed()
  fun onAdOpened()
  fun onAdLoaded(view: View?): Boolean
  fun onAdClicked()
  fun onAdError(errorCode: ErrorCode)
  fun onAdRefreshed(view: View?)
  enum class ErrorCode {
    NO_FILL,
    INTERNAL_ERROR,
    TIMEOUT,
    UNKNOWN,
    BAD_REQUEST
  }
}